server/routes/notification.js

```diff
@@ -0,0 +1,167 @@
+const authMiddleware = require("../middlewares/authMiddleware");
+const ListProductModel = require("../models/listProductModel");
+const NotificationModel = require("../models/notificationModel");
+const UserListModel = require("../models/userListModel");
+const { PRICE } = require("../utils/enums");
+const router = require("express").Router();
+
+//Make the given notification id read
+router.put("/read/:id", authMiddleware, async (req, res) => {
+  try {
+    const { id } = req.params;
+    const { id: user_id } = req.user;
+
+    await NotificationModel.update(
+      { isRead: true },
+      {
+        where: {
+          id,
+          user_id,
+        },
+      }
+    );
+
+    //update returns only a 1 or a 0 so we create another query to show the client
+    const newNotification = await NotificationModel.findOne({
+      where: {
+        id,
+        user_id,
+      },
+    });
+
+    res.send(newNotification);
+  } catch (error) {
+    res.status(500).send({
+      msg: "Server Error",
+    });
+  }
+});
+
+//Create a notification using a product id given inside the body
+//Will be used by a service
+router.post("/price", async (req, res) => {
+  const { productId, title, price, previousPrice } = req.body;
+
+  const promiseArr = [];
+
+  //Find all lists associated with this product
+  const listProds = await ListProductModel.findAll({
+    where: {
+      product_id: productId,
+    },
+  });
+
+  let newNotifs = {};
+
+  //Find the owners associated with the list
+  listProds.forEach((listProd) => {
+    promiseArr.push(
+      UserListModel.findOne({
```

Comment:
this has the potential to generate a lot of queries. You could do `await UserListModel.findAll({where: { id: listProds.map( (listProd) => listProd.id ) }})` just once. Also, I'm pretty sure this should be atomic with your last query. Try adding transactions if possible.